### PR TITLE
tasks pagenation removed

### PR DESCRIPTION
**Pull request by RomanKlymus**
Just remove task pagenation from DTO and some methods

### CODE REVIEW

File: src/main/java/com/softserve/teachua/controller/ChallengeController.java [hunk 1]
```diff
@@ -34,21 +32,21 @@ public ChallengeController(ChallengeService challengeService) {
...
+     * Use this endpoint to get full information about challenge with tasks that have already begun.
      *
      * @param id - put challenge id here.
      * @return {@code ChallengeProfile}.
      */
     @GetMapping("/challenge/{id}")
-    public ChallengeProfile getChallenge(@PathVariable Long id, @PageableDefault(value = 2, sort = "startDate") Pageable pageable) {
-        return challengeService.getChallenge(id, pageable);
+    public ChallengeProfile getChallenge(@PathVariable Long id) {
+        return challengeService.getChallenge(id);
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/controller/ChallengeController.java:42]: removed pageable from params

File: src/main/java/com/softserve/teachua/controller/TaskController.java [hunk 2]
```diff
@@ -28,16 +26,11 @@ public TaskController(TaskService taskService) {
      * This feature available only for admins.
      *
      * @param id - put challenge id here.
-     * @param pageable - put new instance of Pageable interface with objects per page value, and sort param.
-     *                 By default, value = 2, and sort param is startDate/
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/controller/TaskController.java:32]: docs fixed

File: src/main/java/com/softserve/teachua/controller/TaskController.java [hunk 3]
```diff
@@ -28,16 +26,11 @@ public TaskController(TaskService taskService) {
...
      * @return {@code Page<TaskPreview} - all tasks with pagination
      */
     @GetMapping("/challenge/{id}/tasks")
-    public Page<TaskPreview> getTasksByChallenge(
-            @PathVariable Long id,
-            @PageableDefault(value = 2, sort = "startDate") Pageable pageable)
-    {
-        return taskService.getTasksByChallengeId(id, pageable);
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/controller/TaskController.java:40]: remove pageable just like above

File: src/main/java/com/softserve/teachua/dto/challenge/ChallengeProfile.java [hunk 4]
```diff
@@ -25,7 +24,7 @@
     private String picture;
     private Long sortNumber;
     private Boolean isActive;
-    private Page<TaskPreview> tasks;
+    private List<TaskPreview> tasks;
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/dto/challenge/ChallengeProfile.java:27]: Change collection type in DTO

File: src/main/java/com/softserve/teachua/repository/TaskRepository.java [hunk 5]
```diff
@@ -2,18 +2,17 @@
...
+import java.util.List;
 
 @Repository
 public interface TaskRepository extends JpaRepository<Task, Long> {
 
-    Page<Task> findTasksByChallenge(Challenge challenge, Pageable pageable);
+    List<Task> findTasksByChallenge(Challenge challenge);
 
-    Page<Task> findTasksByChallengeAndStartDateBefore(Challenge challenge, LocalDate date, Pageable pageable);
+    List<Task> findTasksByChallengeAndStartDateBefore(Challenge challenge, LocalDate date);
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/repository/TaskRepository.java:16]: remove pageable from repository and change return type to List

File: src/main/java/com/softserve/teachua/service/impl/ChallengeServiceImpl.java [hunk 6]
```diff
@@ -122,14 +119,14 @@ public ChallengeDeleteResponse deleteChallenge(Long id) {
...
         Challenge challenge = getChallengeById(id);
         ChallengeProfile challengeProfile =
                 dtoConverter.convertToDto(challenge, ChallengeProfile.class);
         Function<Task, TaskPreview> function = (task) -> dtoConverter.convertToDto(task, TaskPreview.class);
-        Page<Task> pageableTasks = taskRepository.findTasksByChallengeAndStartDateBefore(challenge, LocalDate.now(), pageable);
-        List<TaskPreview> tasks = pageableTasks.stream().map(function).collect(Collectors.toList());
-        challengeProfile.setTasks(new PageImpl<>(tasks, pageableTasks.getPageable(), pageableTasks.getTotalElements()));
+        List<TaskPreview> tasks = taskRepository.findTasksByChallengeAndStartDateBefore(challenge, LocalDate.now())
+                .stream().map(function).collect(Collectors.toList());
+        challengeProfile.setTasks(tasks);
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/service/impl/ChallengeServiceImpl.java:129]: replace pageable convert to list

File: src/main/java/com/softserve/teachua/repository/TaskRepository.java [hunk 7]
```diff
@@ -13,6 +13,6 @@
 
     List<Task> findTasksByChallenge(Challenge challenge);
 
-    List<Task> findTasksByChallengeAndStartDateBefore(Challenge challenge, LocalDate date);
+    List<Task> findTasksByChallengeAndStartDateBeforeOrderByStartDate(Challenge challenge, LocalDate date);
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/repository/TaskRepository.java:16]: I just added the "order by" command to request

File: src/main/java/com/softserve/teachua/service/impl/ChallengeServiceImpl.java [hunk 8]
```diff
@@ -124,7 +124,8 @@ public ChallengeProfile getChallenge(Long id) {
         ChallengeProfile challengeProfile =
                 dtoConverter.convertToDto(challenge, ChallengeProfile.class);
         Function<Task, TaskPreview> function = (task) -> dtoConverter.convertToDto(task, TaskPreview.class);
-        List<TaskPreview> tasks = taskRepository.findTasksByChallengeAndStartDateBefore(challenge, LocalDate.now())
+        List<TaskPreview> tasks = taskRepository
+                .findTasksByChallengeAndStartDateBeforeOrderByStartDate(challenge, LocalDate.now().plusDays(1))
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/service/impl/ChallengeServiceImpl.java:128]: here I added one day to the current day to get tasks which starts today

File: src/main/java/com/softserve/teachua/controller/ChallengeController.java (hunk 1 above)
Comment (VictoriaAmazonka) [reply src/main/java/com/softserve/teachua/controller/ChallengeController.java:42]: Was there an issue with pageable?

**Review by VictoriaAmazonka — APPROVED** [review]

**Pull request merged**